main.go

```diff
@@ -16,44 +19,65 @@ var (
 	permission = pflag.String("permission", "", "File permission")
 )
 
+func init() {
+	os.Setenv("PROVIDER_LOG_FILE", "/var/log/azure-provider.log")
+}
+
 func main() {
-	pflag.Parse()
+	klog.InitFlags(nil)
 
-	ctx := context.Background()
+	pflag.CommandLine.AddGoFlagSet(goflag.CommandLine)
+	pflag.Parse()
 
-	var attrib map[string]string
-	var secret map[string]string
+	var attrib, secret map[string]string
 	var filePermission os.FileMode
+	var f *os.File
+	var err error
+
+	// setup log file into which provider logs will be written to
+	if f, err = setupLogFile(); err != nil {
+		klog.Fatalf("[error]: %v", err)
+	}
+	defer f.Close()
+	os.Stdout = f
+	os.Stderr = f
 
-	err := json.Unmarshal([]byte(*attributes), &attrib)
+	err = json.Unmarshal([]byte(*attributes), &attrib)
 	if err != nil {
-		log(fmt.Sprintf("failed to unmarshal attributes, err: %v\n", err))
-		os.Exit(1)
+		klog.Fatalf("failed to unmarshal attributes, err: %v", err)
 	}
 	err = json.Unmarshal([]byte(*secrets), &secret)
 	if err != nil {
-		log(fmt.Sprintf("failed to unmarshal secrets, err: %v\n", err))
-		os.Exit(1)
+		klog.Fatalf("failed to unmarshal secrets, err: %v", err)
 	}
 	err = json.Unmarshal([]byte(*permission), &filePermission)
 	if err != nil {
-		log(fmt.Sprintf("failed to unmarshal file permission, err: %v\n", err))
-		os.Exit(1)
+		klog.Fatalf("failed to unmarshal file permission, err: %v", err)
 	}
 
 	provider, err := NewProvider()
 	if err != nil {
-		log(fmt.Sprintf("error creating new provider : %v\n", err))
-		os.Exit(1)
+		klog.Fatalf("[error] : %v", err)
 	}
+
+	ctx := context.Background()
 	err = provider.MountSecretsStoreObjectContent(ctx, attrib, secret, *targetPath, filePermission)
 	if err != nil {
-		log(fmt.Sprintf("error mounting secret store object content : %v\n", err))
-		os.Exit(1)
+		klog.Fatalf("[error] : %v", err)
 	}
+
+	klog.Flush()
 	os.Exit(0)
 }
 
-func log(msg string) {
-	fmt.Fprintf(os.Stdout, msg)
+func setupLogFile() (*os.File, error) {
+	fileName := os.Getenv("PROVIDER_LOG_FILE")
+	if fileName == "" {
```

Comment:
since you set `PROVIDER_LOG_FILE` in `init()`, when will this be true?

Reply:
My initial plan was to add it as an env var while building binary which is why I added the check. With `init` it should never be true.

Reply:
If this is never true, maybe we should remove this?